### DISCUSSION TITLE
Trackball does not work

### DISCUSSION THREAD

**Discussion by iamim** [Yowkees/keyball#429] [Q&A]
I recently finished the keyball39 build. Unfortunately, I almost immediately started having issues with the trackball. At first, it was acting choppy (it would work for a second, then it would stop; sometimes it would react for one jump if the ball in pressed) for the first boot up. Shortly after it stopped working at all.
Because it was reacting to the ball jumping (it happens if I move it under too steep angle) I decided it was because of a bad connection. Today I re-soldered the right-angle conthrough connection to the PCB and it worked. However, after reassembling, it stopped working again.
Because during assembly I would have to rotate the assembly I thought it could also be the sensor and the breakboard connection. I re-soldered that and it didn't help.

It still does not work and I feel very defeated. Please advice on what I could do to fix it.
Maybe I can order a new conthrough+breakboard+sensor assembly or is there a way to debug it?

**Comment by Yowkees** [comment]
Hi. Thank you for choosing Keyball39.

In this case, I think there is a problem on Trackball PCB or L-shaped conthrough.
I want to check soldering of them before trying new parts.

Can you show me the photo of Trackball PCB, conthrough and middle PCB?
Thank you very much.

**Comment by iamim** [comment]
I did some more thinking and remembered that I had to re solder the microcontroller. So I checked it and it turns out one pin wasn’t connecting properly 🙂
Thank you for offering your help!
